colab/test_notebooks.py

```diff
@@ -0,0 +1,54 @@
+# Lint as: python3
```

Comment:
The `run_python_notebook.sh` script is generic, and is also used for [the mixed sample](https://github.com/google/iree/tree/main/iree/samples/variables_and_state). This script is specific to the notebooks in this directory.

I have a few more samples planned that will also put Colab notebooks side by side with C code. In fact, I'm thinking about merging the entire top level `colab/` directory with `iree/samples/`...

Reply:
If we had more notebooks outside of the central `colab/` directory that were _not_ covered by their own tests, then it might make sense to centralize this test script and have it take explicit paths to all notebooks (or grep the entire repo).

For now I'm thinking: this test script for `colab/`, individual test scripts for `iree/samples/...` (variables and state, dynamic shapes, and future samples that use notebooks)

Reply:
(those `samples/` use the outputs of the notebooks, so running the notebooks in isolation isn't sufficient)